app/jobs/attach_remote_file_to_upload_job.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+##
+# Job to upload files from a URL in the background
+class AttachRemoteFileToUploadJob < ApplicationJob
+  queue_as :default
+
+  def perform(upload)
+    upload.files.attach(io: URI.open(upload.url), filename: upload.name)
+  rescue SocketError, OpenURI::HTTPError => e
+    error = "Error opening #{url}: #{e}"
+    Rails.logger.info(error)
+    Honeybadger.notify(error)
+    raise e
```

Comment:
I'm just re-raising here so we retry (incase the issue was a temp. network blip or the external file store is temp. responding w/ an error).

Not sure if we want to do anything more sophisticated now, or wait until we have more data on what kind of issues we're likely to see uploading via URL.